### ISSUE TITLE
Crash when `sidebar` config object is not defined

### ISSUE THREAD

**Issue by FollieHiyuki** [willothy/nvim-cokeline#170]
I don't use the sidebar feature, so my config doesn't have the sidebar config object. This results in cokeline crashing whenever I open NvimTree on the left (the right side is unaffected):

```lua
E5108: Error executing lua ...l/share/nvim/lazy/nvim-cokeline/lua/cokeline/sidebar.lua:146: attempt to index a nil value
stack traceback:
	...l/share/nvim/lazy/nvim-cokeline/lua/cokeline/sidebar.lua:146: in function 'get_components'
	...share/nvim/lazy/nvim-cokeline/lua/cokeline/rendering.lua:107: in function 'prepare'
	...share/nvim/lazy/nvim-cokeline/lua/cokeline/rendering.lua:220: in function <...share/nvim/lazy/nvim-cokeline/lua/cokeline/rendering.lua:219>

```

If I add a non-empty sidebar block, for example:

```lua
    sidebar = {
        filetype = { 'aerial' },
        components = {
            {
                text = function(buf)
                    return ' ' .. buf.filetype
                end,
                fg = colors.yellow,
                bg = function() return require('cokeline.hlgroups').get_hl_attr('NvimTreeNormal', 'bg') end,
                bold = true,
            },
        },
    },
```
cokeline operates normally (even though I don't specify NvimTree as a sidebar filetype).

**Comment by willothy** [comment]
Thanks for reporting, I'll look into this ASAP. Could you provide a minimal init for testing? Thanks!

**Comment by FollieHiyuki** [comment]
This should be enough to reproduce I think.

```lua
require('cokeline').setup {
  components = { },
}
```

**Comment by miversen33** [comment]
@willothy I ran into this issue as well (finally updated my plugins). The problem you are running into is in [sidebar.lua:115-149](https://github.com/willothy/nvim-cokeline/blob/main/lua/cokeline/sidebar.lua#L115-L149)

Specifically this little bit of logic here

```lua
-- Starting at line 138
  if width > sidebar_width then
    sort(sidebar_components, rendering.by_decreasing_priority)
    components.shorten(sidebar_components, sidebar_width)
    sort(sidebar_components, rendering.by_increasing_index)
  elseif width < sidebar_width then
    local space_left = sidebar_width - width
    local last = #sidebar_components
    sidebar_components[last].text = sidebar_components[last].text
      .. rep(" ", space_left)
    sidebar_components[last].width = sidebar_components[last].width
      + space_left
  end
  ```
  This logic assumes that _something_ is in your `sidebar_components`. This assumption is unsafe as the only way anything is in `sidebar_components` is if `states.sidebar` has a length of greater than 0. However looking at the default configuration (found in [config.lua:190-200](https://github.com/willothy/nvim-cokeline/blob/main/lua/cokeline/config.lua#L190-L200)) we can see that the only way state.sidebar is populated is if config.sidebar.components is not **nil** and the default configuration for `nvim-cokeline` indeed specifies that [config.sidebar.components is not nil, rather its an empty table](https://github.com/willothy/nvim-cokeline/blob/main/lua/cokeline/config.lua#L118)
  
  So in short, the default configuration provided by cokeline sets `config.sidebar.components` to an empty table, which does __not__ fail a truthy check (here is a repl example showing this)
![image](https://github.com/willothy/nvim-cokeline/assets/2640668/a07269f6-1c4c-4261-8f9d-c7b3f36dfd27)

  
  Because this empty table is truthy, `state.sidebar` is never populated with _something_ and thus you get a `nil index` error at `sidebar.lua:146` due to attempting to index an empty table and concat a string to a nil attribute `sidebar_components[last].text .. rep(" ", space_left)`

**Comment by miversen33** [comment]
#177 addresses this issue. Tested with no sidebar (the default configuration provided by cokeline) and with a set of custom sidebar components

**Comment by willothy** [comment]
Thanks for the PR! Sorry for not getting to this, I've been a bit busy recently.

**Comment by willothy** [comment]
Fixed in #177

**Comment by miversen33** [comment]
> Thanks for the PR! Sorry for not getting to this, I've been a bit busy recently.

You owe us nothing :) thanks for picking up this project, maintaining and improving it! I'm glad I could help

**Comment by willothy** [comment]
Totally, thank you! I appreciate the contributions :)
